### PR TITLE
Modernized Windows Support and documentation update

### PR DESCRIPTION
**Pull request by yakatz**
Three parts to this:

1. Support for generating Windows facts automatically
2. The Windows facts themselves
3. Rewriting the documentation for fact generation with the aim of ensuring repeatable builds

### CODE REVIEW

**Comment by yakatz** [comment]
The tests fail because none of the new files contain legacy facts. I saw the discussion on #266 (which has been open for a year). Should I bother adding them or will that change also happen?

**Comment by zilchms** [comment]
Looks good to me on principle. One question though: Do you happen to know why we are loosing some facts for the already present windows-10-facts for example? Are those the missing legacy facts you pointed out?

File: facts/4.3/windows-10-x86_64.facts [hunk 1]
```diff
@@ -1,120 +1,88 @@
...
-  "network6": "fe80::",
-  "network6_Ethernet": "fe80::",
-  "network_Ethernet": "10.138.1.0",
   "networking": {
-    "dhcp": "168.63.129.16",
-    "domain": "example.com",
-    "fqdn": "foo.example.com",
+    "dhcp": "10.0.2.2",
+    "domain": "exmaple.com",
+    "fqdn": "foo.exmaple.com",
```
Comment (zilchms) [inline facts/4.3/windows-10-x86_64.facts:44]: This breaks at least one test and seems like a typo, which is wierd since it is fully autogenerated

Comment (yakatz) [reply facts/4.3/windows-10-x86_64.facts:44]: Woops, that was my typo. Regenerating...

**Comment by yakatz** [comment]
> The tests fail because none of the new files contain legacy facts. I saw the discussion on #266 (which has been open for a year). Should I bother adding them or will that change also happen?

Added them back

**Review by zilchms — APPROVED** [review]

**Comment by yakatz** [comment]
Rebased on current master

File: facts/windows_get_facts.ps1 [hunk 2]
```diff
@@ -1,10 +1,29 @@
 param(
-  [array]$puppetAgentVersions = ('1.1.1','1.2.2','1.4.0'),
-  [string]$baseUrl = 'https://downloads.puppetlabs.com/windows',
+  [array]$puppetAgentVersions = ('7.24.0','8.0.0','8.5.1'),
```
Comment (bastelfreak) [inline facts/windows_get_facts.ps1:2]: could you add `8.6.0`?

File: facts/README.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+## Legacy Facts
+
+*NOTE*: When using Facter version 4, by default some "legacy facts" are hidden from the output.
+To generate a fact set with the legacy facts use the command `puppet facts show --show-legacy`
+
+## OS-Specific Notes
+
+### Ubuntu 18.04
+
+The default Ubuntu 18.04 `box` creates a console log.
```
Comment (bastelfreak) [inline facts/README.md:87]: what kind of log do you mean? I don't understand that phrase.

File: facts/windows_get_facts.ps1 (hunk 2 above)
Comment (yakatz) [reply facts/windows_get_facts.ps1:2]: Coming right up

File: facts/README.md (hunk 3 above)
Comment (yakatz) [reply facts/README.md:87]: The Vagrant box creates a console log. It looks like I never actually commit the code to disable it, so I will just take that note out.

Comment (yakatz) [reply facts/README.md:87]: I just wanted to have a note about something besides Windows. I made it more understandable.

**Comment by yakatz** [comment]
@bastelfreak All the changes you requested are done

**Review by bastelfreak — APPROVED** [review]

**Pull request merged**